portfolio/src/main/webapp/script.js

```diff
@@ -155,64 +179,111 @@ function createComment(comment) {
         aria-labelledby="rec-${comment.id}"
       >
         <div class="rec-description">
-          <div class="rec-description-txt">
-            ${comment.description || 'No Description.'}
+          <div class="rec-description-txt">${
+            comment.description || 'No Description.'
+          }</div>
+          <div class="row">
+            <a
+              href=${comment.link}
+              target="_blank"
+              rel="noopener noreferrer"
+              class=${linkClass}
+              >Learn more</a
+            >
+            <i class="fa fa-ban col-2" id="delete-btn-${comment.id}"></i>
           </div>
-          <a
-            href=${comment.link}
-            target="_blank"
-            rel="noopener noreferrer"
-            class=${linkClass}
-            >Learn more</a
-          >
         </div>
       </div>
     </div>
   `);
 }
 
 /**
- * fetch text from /data to display
+ * Add event listener to detect when comment is being collapsed/expanded in order to rotate caret accordingly
+ * @param {string} cid comment id used to distinguish each unique comment
  */
-function getFetchRequest() {
+function addRotationEvent(cid) {
+  const $elem = $(`#${cid}`);
+  $elem.on('click', () => {
+    $elem.find('.fa').toggleClass('rotated');
+    const angle = $elem.find('.fa').hasClass('rotated') ? 90 : 0;
+    $elem.find('.fa').animate(
+      { deg: angle },
+      {
+        duration: 200,
+        step: function (now) {
+          $elem.find('.fa').css({
+            transform: 'rotate(' + now + 'deg)',
+          });
+        },
+      }
+    );
+  });
+}
+
+/**
+ * Delete the comment selected from datastore, reload existing comments
+ * @param {string} cid comment id used to distinguish each unique comment
+ */
+function deleteComment(cid) {
+  const id = getCommentId(cid);
+  $.ajax({
```

Comment:
This is fine, but if you want to be consistent I'd either use this instead of fetch, or use 
```javascript
await fetch(`/delete-data?id=${id}`, {method: 'POST'});
```

Reply:
See https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch